first_hour_on_vwb/working_with_data.ipynb

```diff
@@ -0,0 +1,577 @@
+    "BUCKET_REFERENCE = \"ws_files_autodelete_after_two_weeks\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "MY_BUCKET = get_bucket_url_from_reference(BUCKET_REFERENCE)\n",
+    "print(f'Bucket ID: {MY_BUCKET}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Working with data from referenced resources\n",
+    "\n",
+    "<a id='access-ref-data'></a>\n",
+    "\n",
+    "[Referenced resources](https://terra-docs.api.verily.com/docs/getting_started/web_ui/#referenced-vs-workspace-controlled-resources) represent data or other elements in Verily Workbench by pointing to a source that exists outside of the current workspace. To add a referenced resource to your workspace, use the <a href=\"../working_with_resources.ipynb\">working_with_resources.ipynb</a> notebook."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Reading data from BigQuery\n",
+    "<a id='read-from-bq'></a>\n",
+    "\n",
+    "There are many ways to interact with your referenced resource data in a Verily Workbench cloud environment. This notebook provides examples of several options, each of which are appropriate for certain use cases."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using cell magics\n",
+    "<a id='cell-magics'></a>\n",
+    "\n",
+    "For troubleshooting, demos and developing new queries, it's useful to leverage the [IPython magics for BigQuery](https://cloud.google.com/python/docs/reference/bigquery/latest/magics), several examples of which are provided below.\n",
+    "\n",
+    "<div class=\"alert alert-block alert-success\">\n",
+    "<b>Note:</b> \n",
+    "    The examples in this sub-section utilize data from the 1000 Genomes data collection. Please add the BigQuery tables from that data collection to your current workspace before running the cells below. Instructions for adding a data collection to your workspace can be found <a href=\"https://support.workbench.verily.com/docs/how_to_guides/work_with_data/#add-a-data-collection\">here</a>.</div>\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "#### Getting summary stats\n",
+    "\n",
+    "<a id='inspect-bq-stats'></a>\n",
+    "\n",
+    "To get summary statistics and visualizations for all the columns of a BigQuery table, you can use the `%bigquery_stats` cell magic. Run the cell below to view summary stats for all the table columns in the pedigree table of the 1000 Genomes dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%bigquery_stats bigquery-public-data.human_genome_variants.1000_genomes_pedigree"
```

Comment:
Cool, I wasn't aware of bigquery_stats. Do you know if this is a costly query to run? I notice that it takes significantly longer than some of the other queries to run.

Reply:
I'm not exactly sure how much it costs. Looking at the [logs](https://screenshot.googleplex.com/4Tw7KH4JiEtuwAR), this query takes about 50 seconds to run, so it does take longer than the other queries in the notebook. However, it is running several queries per column (finding # observations, # distinct, # missing, % distinct, % missing, and plotting the frequency of each possible value) for each of the 12 columns in this particular table, I think it's not necessarily unreasonable that it takes this long. I could include a caveat that running this cell may take about a minute and could be expensive relative to other queries. What do you think?

Reply:
Can we at least confirm and document the size of the dataset?